src/redux/actions/actionsSlice.ts

```diff
@@ -73,23 +73,19 @@ export const actionsSlice = createSlice({
       };
     },
     removeAction: (state, action: PayloadAction<string>) => {
-      const newData = state.actions;
-      const index = newData.findIndex((element) => element.identifier === action.payload);
+      const index = state.actions.findIndex(
+        (element) => element.identifier === action.payload
+      );
       if (index > -1) {
         // only splice array when item is found
-        newData.splice(index, 1); // 2nd parameter means remove one item only
+        state.actions.splice(index, 1); // 2nd parameter means remove one item only
       }
       // If the removed action was the last one with its gui path, we need to change the
       // navigation path
-      const indexPath = newData.findIndex(
+      const indexPath = state.actions.findIndex(
         (element) => element.guiPath === state.navigationPath
       );
-      const newNavigationPath = indexPath < 0 ? '/' : state.navigationPath;
-      return {
-        ...state,
-        data: [...newData],
-        navigationPath: newNavigationPath
-      };
+      state.navigationPath = indexPath < 0 ? '/' : state.navigationPath;
```

Comment:
```suggestion
      if (indexPath === -1) {
        state.navigationPath = '/';
      }
```
 Checking for -1 explicitly is clearer here, since that's the expected return value. And we don't need to set the value if that was not the case